### DISCUSSION TITLE
Understanding the "Loop Once" feature of scheduled playlist.

### DISCUSSION THREAD

**Discussion by radioindradhanush** [AzuraCast/AzuraCast#5634] [General]
If my scheduled playlist contains just one 12-second long voice file (an announcement), which I want to play just once at the scheduled time, should I click "Loop Once" setting for the playlist or should I leave this option unchecked? And if, suppose, I have to play it at 8:00 AM, should the "start-time" and "end-time" be entered as **8:00 AM** and **8:00 AM** OR **8:00 AM** and **8:01** AM, respectively, as the **duration of the file is just 12 seconds**?

**Comment by Mozitux** [comment]
Hi, for my part, I never paid attention to this option. 

I have playlists containing only one sound, each about 10 to 30 seconds long and even without this option, the playlist is played correctly at the scheduled time, sometimes with a little advance or delay depending on the duration of current titles.

**Reply by radioindradhanush** [reply]
Hi, thanks for your response.

Could you please share your settings with me? I have to play a 12-second long sound (an announcement) at **6:15 AM and 7:45 PM daily.** Should the "start-time" and "end-time" in two Schedules be entered as **_6:15 AM and 6:15 AM_**, and **_7:45 PM and 7:45 PM_** OR **_6:15 AM and 6:16 AM_** and **_7:45 PM and 7:46 PM_** respectively? Screen captures of **_my settings_** for the playlist containing this 12-second long sound (announcement) are posted in this message. Do you also have the same settings, or are your settings different from mine?

Warm regards.

![Playlist Settings - 1](https://user-images.githubusercontent.com/110247251/182272565-2e4e1677-accf-44f5-b6f1-f5f93e580283.png)
![Playlist Settings - 2](https://user-images.githubusercontent.com/110247251/182272588-d931ae00-ddd3-4eee-b088-a75f36fe3f81.png)
![Playlist Settings - 3](https://user-images.githubusercontent.com/110247251/182272619-584e242f-7820-492b-944f-90519d168e30.png)
![Playlist Settings - 4](https://user-images.githubusercontent.com/110247251/182272641-ba3e8d10-e4bb-43ea-ba1b-6f16347f6cc7.png)

**Reply by Mozitux** [reply]
Of course, here's my settings for a Jingle that plays every weeknight at 10:00 p.m. (Sorry, my station is in French)
As you can see, my start time and end time is the same, and it's work without any problem for my part.

![01](https://user-images.githubusercontent.com/19860756/182274345-ebc31981-6a05-4f67-a90b-02c1c493cdab.PNG)
![02](https://user-images.githubusercontent.com/19860756/182274346-b5654dbb-9f9e-41e3-a3fb-43f25297ef40.PNG)
![03](https://user-images.githubusercontent.com/19860756/182274339-017bc923-4f88-4f1a-ae2f-f6a126d8356c.PNG)

**Reply by jayp89** [reply]
Can someone tell me why when I schedule a playlist with one or two tracks in it like for example if I have a new segment which is pre-recorded and is uploaded every day how would I get the relevant episode for today to play today in full currently it's only playing basically about 3 minutes of a 7 minute file

**Reply by Mozitux** [reply]
Can you provide screenshots of your configuration?

**Comment by radioindradhanush** [comment]
Thank you for sharing your settings!

I can see the differences. Firstly, your jingle plays **only once a day**, whereas my announcement plays **two times a day**. Secondly, you have given **_weight 25,_** whereas I've given **_weight 4_** to my playlist containing single file. Thirdly, I've unchecked the days (Monday, Tuesday, etc.), whereas you have checked all of them though I don't this should make any difference.

**Reply by Mozitux** [reply]
Don't worry, repeating them several times a day does not change the operating principle, and you can also check only the days that interest you. For my part, I needed for this specific playlist that it plays once a day, every day of the week.

Here's another example setup I have for a newscast that airs on Sunday and Monday at two different times:
![04](https://user-images.githubusercontent.com/19860756/182409483-376c7526-52d3-4bea-b12c-4ad45004d4b3.PNG)

For the weight of the playlist, it must be greater than that of the previous playlist in order to be sure that it will be read in priority.

**For example :**
Music Playlist > Weight of 3
Jingle Playlist > Higher than 3
Music Playlist > Back to weight of 3

**Reply by radioindradhanush** [reply]
Thank you for further clarification. Which version of Azuracast are you using?

The problem I'm facing is that even after assigning weight of 6 to the playlisit containing the announcement which I want to be played twice everyday at 6:15 AM and 7:45 PM is not playing even once. Just now, I've increased it's weight to 8 (I noticed, you had assigned the weight of 25). I hope it works. 

Because I want this announcement to be played everyday, I've not checked any of the boxes seen next to the names of days on the Scheduling page though I noticed that you had checked each of them in the screenshots you shared with me earlier. If assigning highr weight doesn't work, I'll check all the boxes seen next to the names of days.

By the way, I see that in the screenshot you've just shared with me, the Start Time is set to be 20:25 PM and the End Time is set to be 20:30 on Sunday. Does this setting work even if the duration of sound/voice to be played is less than 1 minute?

**Reply by Mozitux** [reply]
For the announcement to be played every day of the week, it is necessary to check every day of the week.

The setting works even if your sound is only 5 seconds long, just as I said above, it may not play at the exact second of the scheduled time, because Azuracast leaves the time to the current file to end.

I'm using the latest stable version of Azuracast with Docker. :)

**Reply by radioindradhanush** [reply]
Thanks for the clarification. Please also confirm whether you need to "Restart" the radio station after saving the scheduling changes?

**Reply by Mozitux** [reply]
There is no need to restart the station when adding music or making changes to playlists.
It is only if you change parameters in your station or in your mount point that you will need to restart it.

**Comment by radioindradhanush** [comment]
Thanks a lot.

**Reply by Mozitux** [reply]
No problem, have fun with Azuracast. 😃

**Comment by ivellios1988** [comment]
If I'm correct, with 'loop once' checked my playlist will play two times in a row, whereas unchedked means my playlist should play only once? Because with 'loop once' checked I observe that my playlists are played at least two times in a row. I play very long podcasts in my radio, with durations reaching even five hours, and I observe that once an episode is over it starts playing again immediately. I'm using AzuraCast v0.17.4 Stable.
'Interrupt other songs' and 'Play only one song' are checked.
